File: app/src/main/java/com/zxn/fuzzysearchdemo/FuzzySearchAdapter.java
package com.zxn.fuzzysearchdemo;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.zxn.fuzzysearch.FuzzySearchBaseAdapter;
import com.zxn.fuzzysearch.IFuzzySearchRule;

import java.util.List;

public class FuzzySearchAdapter extends FuzzySearchBaseAdapter<ItemEntity, FuzzySearchAdapter.ItemHolder> {

	public FuzzySearchAdapter() {
		super(null);
	}

	public FuzzySearchAdapter(IFuzzySearchRule rule) {
		super(rule);
	}

	public FuzzySearchAdapter(List<ItemEntity> dataList) {
		super(null, dataList);
	}

	public FuzzySearchAdapter(IFuzzySearchRule rule, List<ItemEntity> dataList) {
		super(rule, dataList);
	}

	@NonNull
	@Override
	public ItemHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		return new FuzzySearchAdapter.ItemHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_adapter, parent, false));
	}

	@Override
	public void onBindViewHolder(@NonNull ItemHolder holder, int position) {
		holder.mTextName.setText(mDataList.get(position).getValue());
	}

	static class ItemHolder extends RecyclerView.ViewHolder {

		TextView mTextName;

		ItemHolder(View itemView) {
			super(itemView);
			mTextName = itemView.findViewById(R.id.text_item_name);
		}
	}

}

File: app/src/main/java/com/zxn/fuzzysearchdemo/FuzzySearchAdapter.kt
package com.zxn.fuzzysearchdemo

import android.view.View
import android.widget.TextView
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.zxn.fuzzysearch.FuzzySearchBaseAdapter
import com.zxn.fuzzysearch.IFuzzySearchRule
import com.zxn.fuzzysearchdemo.FuzzySearchAdapter.ItemHolder

class FuzzySearchAdapter : FuzzySearchBaseAdapter<ItemEntity, ItemHolder> {

    constructor(dataList: MutableList<ItemEntity>?) : super(null, dataList, R.layout.item_adapter)

    constructor(rule: IFuzzySearchRule?, dataList: MutableList<ItemEntity>?) : super(rule, dataList, R.layout.item_adapter)

    override fun convert(helper: ItemHolder, item: ItemEntity) {
        val mTextName = helper.getView<TextView>(R.id.text_item_name)
        mTextName.text = item.value
    }

    class ItemHolder(itemView: View?) : BaseViewHolder(itemView!!)
}